src/app/logic/player.js

```diff
@@ -0,0 +1,44 @@
+export class Player {
+  constructor() {
+    this.correctAnswersScore = 0;
+    this.amountOfAnsweredQuestions = 0;
+    this.detailQuestionData = [];
+    this.endGameData = {};
+  }
+
+  registerAnswer(correctAnswer, playerAnswer, isanswerCorrect) {
+    if (this.playerAnswer === false) {
+      throw new Error(
+        'error with playerAnswer, playerAnswer is null or undefined',
+      );
+    }
+    if (isanswerCorrect) {
```

Comment:
If you are getting both correct and players answer in here, this seems reduntant. So my suggestion would be to either not pass the correct answer (only players answer and boolean flag true/false if it was correct) and get rid of all the quiz data from the player. Or, if you decide to keep the quiz data in the player, get only player's answr and correct answer and write the line:15 as `if(correctAnswer === playerAnswer) {`

Reply:
But then, this comparison won't be done three time for one question? First time, with correctAnswer function, when boolean is sended back to gameMaker to show at UI proper color. Then second time to check if counter should goes up, and third time when results are shown to the player in table?
Here it will be checked once by correctAnswer function, and result will be send further.
But I'm not sure with option takes less time and memory

Reply:
Well, 3 times is definitely bad! Perhaps you could think of a way not to check it that many times? Right after the check if its correct, try to store it in a way its accessible and perhaps not repeated anywhere else. So maybe the deitaled answets would hold an object `{playerAnswer: 'Harry Portier', wasCorrect: true}`. After the check you send info to view to highlight with the right color, then the view doesnt need it anymore. But a player could hold this info, withuoth the correct answer, the correct answer and imaga data could be stored in the game maker.

Reply:
So, data to register in Player would be that object {playerAnswer: 'Harry Portier', wasCorrect: true}?
and at the end of game, when full data are needed, gameMaker will take array of that objects from Player and array with correct answers from itself, and then combine them together, to give complex info to controller and then view?